infrastructure/quick-deploy/aws/ecr.tf

```
@@ -1,12 +1,12 @@
 locals {
-  mongodb_image_name = can(coalesce(var.mongodb_sharding)) ? coalesce(var.mongodb.image_name, "bitnami/mongodb-sharded") : coalesce(var.mongodb.image_name, "bitnami/mongodb")
+  mongodb_image_name = var.mongodb_sharding != null ? (var.mongodb != null ? coalesce(var.mongodb.image_name, "bitnami/mongodb-sharded") : "bitnami/mongodb-sharded") : (var.mongodb != null ? coalesce(var.mongodb.image_name, "bitnami/mongodb") : "bitnami/mongodb")
```

You can simplify this by leveraging `try`:

```suggestion
  mongodb_image_name = try(coalesce(var.mongodb.image_name), var.mongodb_sharding != null ? "bitnami/mongodb-sharded" : "bitnami/mongodb")
```